DB Tables not Properly Renamed During Installation

[Src downloaded 2014_07_10 from](https://github.com/kimai/kimai/releases/download/v0.9.3-rc.1/kimai_0.9.3.rc1.zip)

Steps that led to the following citation:
_"Error during installation!
Kimai is unable to create necessary tables because tables with the name name already exist. To update your existing installation please click here. Choose another table prefix or delete existing tables with this prefix in order to allow Kimai to perform a clean installation."_
### step_1)

During installation use a huge database table name prefix.

```
kimai_<about 100 characters>
```
### step_2)

Push button "Proceed" and enjoy the error message that says that the table name prefix should be about 64 chars max.
### step_3)

Edit the table name prefix by deleting characters from its end, e.g. by leaving the head part of the text the same.
### step_4)

Push button "Proceed" and enjoy the previously cited error message. :-)

@simonschaufi I got some more information:
Searching most actuaI files after the broken installrion I found the kimai/temporary/logfile.txt that has the following prosa:
[20.01.2018 18:26:22] -- begin install ----------------------------------
[20.01.2018 18:26:22] [ERROR] in [CREATE TABLE `kimai_preferences` (`userID` int(10) NOT NULL,`option` varchar(255) NOT NULL,`value` varchar(255) NOT NULL,PRIMARY KEY (`userID`,`option`));] => s:63:"Specified key was too long; max key length is 767 bytes (#1071)";
[20.01.2018 18:26:23] [ERROR] in [CREATE TABLE `kimai_configuration` (`option` varchar(255) NOT NULL,`value` varchar(255) NOT NULL,PRIMARY KEY(`option`));] => s:63:"Specified key was too long; max key length is 767 bytes (#1071)";
[20.01.2018 18:26:24] [ERROR] in [INSERT INTO `kimai_preferences` (`userID`,`option`,`value`) VALUES('309443751', 'ui.rowlimit', '100'),('309443751', 'ui.skin', 'standard'),('309443751', 'ui.showCommentsByDefault', '0'),('309443751', 'ui.hideOverlapLines', '1'),('309443751', 'ui.showTrackingNumber', '1'),('309443751', 'timezone', 'Europe/Berlin');] => s:53:"Table 'kimai.kimai_preferences' doesn't exist (#1146)";
[20.01.2018 18:26:24] [ERROR] in [INSERT INTO `kimai_configuration` (`option`, `value`) VALUES('version', '1.2.1'),('login', '1'),('adminmail', 'admin@example.com'),('loginTries', '3'),('loginBanTime', '900'),('revision', '1393'),('currency_name', 'Euro'),('currency_sign', '€'),('currency_first', '0'),('show_update_warn', '1'),('check_at_startup', '0'),('show_daySeperatorLines', '1'),('show_gabBreaks', '0'),('show_RecordAgain', '1'),('show_TrackingNr', '1'),('date_format_0', 'dd.mm.yy'),('date_format_1', '%d.%m.'),('date_format_2', '%d.%m.%Y'),('date_format_3', 'd.m.Y'),('table_time_format', '%H:%M'),('language', 'de'),('roundPrecision', '0'),('decimalSeparator', ','),('durationWithSeconds', '0'),('exactSums', '0'),('defaultVat', '0'),('editLimit', '0'),('roundTimesheetEntries', '0'),('roundMinutes', '0'),('roundSeconds', '0'),('allowRoundDown', '0'),('defaultStatusID', '1')] => s:55:"Table 'kimai.kimai_configuration' doesn't exist (#1146)";
[20.01.2018 18:26:24] -- showing install error --------------------------
l

--- a/includes/version.php
+++ b/includes/version.php
@@ -1,4 +1,4 @@
 <?php
-$kga['version'] = '1.2.1';
-$kga['revision'] = 1393; // database revision number (incremented whenever the database changes)
+$kga['version'] = '1.2.2';
+$kga['revision'] = 1394; // database revision number (incremented whenever the database changes)
 $kga['status'] = 'dev'; // leave blank if stable

--- a/installer/install.php
+++ b/installer/install.php
@@ -93,7 +93,7 @@ exec_query($query);
 
 $query = "CREATE TABLE `${p}preferences` (
   `userID` int(10) NOT NULL,
-  `option` varchar(255) NOT NULL,
+  `option` varchar(190) NOT NULL,
   `value` varchar(255) NOT NULL,
   PRIMARY KEY (`userID`,`option`)
 );";
@@ -224,7 +224,7 @@ $query = "CREATE TABLE `${p}timeSheet` (
 exec_query($query);
 
 $query = "CREATE TABLE `${p}configuration` (
-  `option` varchar(255) NOT NULL,
+  `option` varchar(190) NOT NULL,
   `value` varchar(255) NOT NULL,
   PRIMARY KEY  (`option`)
 );";

--- a/installer/processor.php
+++ b/installer/processor.php
@@ -188,7 +188,7 @@ switch ($axAction) {
         $result = false;
         $config = new Kimai_Config(array());
 
-        $database = new Kimai_Database_Mysql(, false);
+        $database = new Kimai_Database_Mysql($config, false);
         $database->connect($hostname, null, $username, $password, true);
         $conn = $database->getConnectionHandler();
 

--- a/installer/processor.php
+++ b/installer/processor.php
@@ -160,7 +160,7 @@ switch ($axAction) {
         $timezone = $_REQUEST['timezone'];
 
         $kimaiConfig = new Kimai_Config(array(
-            'server_prefix' => $server_prefix,
+            'server_prefix' => $prefix,
             'server_hostname' => $hostname,
             'server_database' => $database,
             'server_username' => $username,
@@ -186,8 +186,9 @@ switch ($axAction) {
 
         $db_error = false;
         $result = false;
+        $config = new Kimai_Config(array());
 
-        $database = new Kimai_Database_Mysql($result, false);
+        $database = new Kimai_Database_Mysql(, false);
         $database->connect($hostname, null, $username, $password, true);
         $conn = $database->getConnectionHandler();
 

--- a/updater/updater.php
+++ b/updater/updater.php
@@ -1174,6 +1174,12 @@ if ((int)$revisionDB < 1393) {
     exec_query("ALTER TABLE `${p}timeSheet` CHANGE `fixedRate` `fixedRate` DECIMAL(10,2) NULL");
 }
 
+if ((int)$revisionDB < 1394) {
+    Kimai_Logger::logfile("-- update to r1394");
+    exec_query("ALTER TABLE `${p}preferences` CHANGE `option` `option` VARCHAR(190) NULL");
+    exec_query("ALTER TABLE `${p}configuration` CHANGE `option` `option` VARCHAR(190) NULL");
+}
+
 // ================================================================================
 // FINALIZATION: update DB version number
 // ================================================================================

